controllers/volumereplication_controller.go

```diff
@@ -64,3 +86,34 @@ func (r *VolumeReplicationReconciler) SetupWithManager(mgr ctrl.Manager, cfg *co
 		For(&replicationv1alpha1.VolumeReplication{}).
 		Complete(r)
 }
+
+func (r VolumeReplicationReconciler) getVolumeHandle(instance *replicationv1alpha1.VolumeReplication) (string, error) {
+	pvcName := instance.Spec.DataSource.Name
+	pvc := &corev1.PersistentVolumeClaim{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: pvcName, Namespace: instance.Namespace}, pvc)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.Log.Info("pvc %q not found", pvcName)
+			return "", err
+		}
+		return "", err
+	}
+	// Validate PVC in bound state
+	if pvc.Status.Phase != corev1.ClaimBound {
+		return "", fmt.Errorf("pvc %q is not bound to any pv", pvcName)
+	}
+
+	// Get PV object for the PVC
+	pvName := pvc.Spec.VolumeName
+	pv := &corev1.PersistentVolume{}
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: pvName, Namespace: instance.Namespace}, pv)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.Log.Info("pv not found")
+			return "", err
+		}
+		return "", err
+	}
+
```

Comment:
return PV and PVC objects from this function. if we need more parameters from PVC and PV we don't get to get it from the kubernetes API server again.

Reply:
returning pv and pvc objects.